### ISSUE TITLE
No back moving in search line

### ISSUE THREAD

**Issue by shadowxnex** [dvorka/hstr#247]
It would be convenient if user can moving back from last edit character when typing searching line.
Use case: you type the begin part of regex:

> ^find ][

And then you think of last part, but you don't precisely known where it has. You type:

> ^find $][

Then you need to back at 1 character for type this:

> ^find ][.*$

And your expression:

> ^find ][some_dir_of_searching_file.*$

Then you can get most similar variants on result list:

> find -exec /some_dir_of_searching_file/file
> find -L -. -xtype f -maxdepth 5 -exec /some_dir_of_searching_file/file
> find -L -. -xtype f -executable -exec /some_dir_of_searching_file/file

Then choose one of it.

BUT! You can't move caret from last editing character to do so. You need to remove all last symbols from the place where you want to paste new ones.

**Comment by dvorka** [comment]
Makes sense. This enhancement is related to #226. @shadowxnex thank you for nice user stories!
